packages/context/src/context.ts

```diff
@@ -33,14 +33,22 @@ export class Context {
    * Create a binding with the given key in the context. If a locked binding
    * already exists with the same key, an error will be thrown.
    *
-   * @param key Binding key
+   * @param keyOrBinding Binding key or a binding
    */
-  bind(key: string): Binding {
+  bind(keyOrBinding: string | Binding): Binding {
```

Comment:
I am not very happy about this method overload. So far, `bind` is always followed with `.to{something}()` and possibly other modifiers like `inScope`, etc. Do you expect people will call those modifiers when binding a Binding directly? I don't.

Instead of overloading the existing `bind` method, I am proposing to add a new method called `add`. That way we have to distinctive usage patterns:

```ts
ctx.bind('foo').to('bar').inScope(BindingScope.SINGLETON);

const b = new Binding('foo').to('bar').inScope(BindingScope.SINGLETON);
ctx.add(b);
```